src/ConnectionPool.cpp

```diff
@@ -70,6 +70,29 @@ int ConnectionPool::init(const char* const * hosts, const uint32_t* ports, const
   return rv;
 }
 
+int ConnectionPool::updateServers(const char* const* hosts, const uint32_t* ports, const size_t n,
+                                   const char* const* aliases) {
+  int rv = 0;
+  if (m_nConns != n) {
+    return 1;
+  }
+  for (size_t i = 0; i < m_nConns; i++) {
+    if (m_conns[i].hasAlias() && (strcmp(m_conns[i].name(), aliases[i]) != 0)) {
+      return 2;
+    }
+  }
+  for (size_t i = 0; i < m_nConns; i++) {
+    if ((strcmp(m_conns[i].host(), hosts[i]) == 0) && (m_conns[i].port() == ports[i])) {
+      rv += -1;
```

Comment:
--rv;

as https://github.com/douban/libmc/pull/61/commits the first commit of this pr try to ensure all unary incr/decr to use this form except those in for statement。